google-cloud-firestore/CONTRIBUTING.md

```diff
@@ -1,187 +1,390 @@
+   for your project, you will need to use a different project for Firestore.
+
+1. Follow the instructions for [Creating a Service Account](AUTHENTICATION.md#creating-a-service-account) in
+   `AUTHENTICATION.md`, including downloading and securely storing a JSON key file. 
+
+1. Set the `GCLOUD_TEST_KEYFILE` environment variable to the path of the JSON key file that you downloaded in the
+   previous step:
+
+   ``` sh
+   $ export GCLOUD_TEST_KEYFILE=/path/to/keyfile.json
+   ```
+
+   If you are already using the `GCLOUD_TEST_KEYFILE` environment variable, and wish to test this library with a
+   different key file, you may set the `FIRESTORE_TEST_KEYFILE` environment variable instead:
+
+   ``` sh
+   $ export FIRESTORE_TEST_KEYFILE=/path/to/keyfile.json
+   ```
+
+1. Set the `GCLOUD_TEST_PROJECT` environment variable to your Google Cloud project ID:
 
-## Firestore Tests
+   ``` sh
+   $ export GCLOUD_TEST_PROJECT=my-project-id
+   ```
 
-Tests are very important part of google-cloud-firestore. All contributions
-should include tests that ensure the contributed code behaves as expected.
+   If you are already using the `GCLOUD_TEST_PROJECT` environment variable, and wish to test this library with a
+   different project, you may set the `FIRESTORE_TEST_PROJECT` environment variable instead:
 
-To run the unit tests, documentation tests, and code style checks together for a
-package:
+   ``` sh
+   $ export FIRESTORE_TEST_PROJECT=my-project-id
+   ```
+
+To set up your interactive console credentials:
+
+1. Set the `GOOGLE_APPLICATION_CREDENTIALS` environment variable to the path of your service account JSON key file (see
+   above):
+
+   ``` sh
+   $ export GOOGLE_APPLICATION_CREDENTIALS=/path/to/keyfile.json
+   ```
+
+   If you are already using the `GOOGLE_APPLICATION_CREDENTIALS` environment variable, and wish to test this library
+   with a different key file, you may set the `FIRESTORE_CREDENTIALS` environment variable instead:
+
+   ``` sh
+   $ export FIRESTORE_CREDENTIALS=/path/to/keyfile.json
+   ```
+
+1. Set the `GOOGLE_CLOUD_PROJECT` environment variable to your Google Cloud project ID:
+
+   ``` sh
+   $ export GOOGLE_CLOUD_PROJECT=my-project-id
+   ```
+
+   If you are already using the `GOOGLE_CLOUD_PROJECT` environment variable, and wish to test this library with a
+   different project, you may set the `FIRESTORE_PROJECT` environment variable instead:
+
+   ``` sh
+   $ export FIRESTORE_PROJECT=my-project-id
+   ```
+
+
+## Run CI
+
+You are now ready to run local CI checks for the library, which you should do **before** you make any changes. Doing so
+ensures that everything is OK with your local environment and the latest dependency versions. You don't want any
+surprises later.
+
+To run the code style checks, documentation tests, and unit tests together, use the `ci` task:
 
 ``` sh
-$ cd google-cloud-firestore/
```

Comment:
I think we should keep this (or alternatively say in prose that you should run these commands within the `google-cloud-firestore` directory) otherwise people may mistakenly run it in the root directory.

Reply:
I'll put it in the prose if that's OK. There are so many examples of commands executed in the `google-cloud-firestore` directory that it looks strange when all of them begin with the `cd`.

Reply:
Added this back at the top of the H2 sections.